cli/src/ui.rs

```diff
@@ -132,7 +132,9 @@ impl UiOutput {
     fn new_paged(pager_cmd: &CommandNameAndArgs) -> io::Result<UiOutput> {
         let mut cmd = pager_cmd.to_command();
         tracing::info!(?cmd, "spawning pager");
-        let mut child = cmd.stdin(Stdio::piped()).spawn()?;
+        // LESS=FRX causes escape sequences to render correctly when PAGER=less
+        // It's harmless when using other pagers
+        let mut child = cmd.env("LESS", "FRX").stdin(Stdio::piped()).spawn()?;
```

Comment:
This was discussed recently, but I think `$LESS` should be set only if it wasn't sent by the current environment nor config.
https://github.com/martinvonz/jj/pull/3657#discussion_r1833657841

Reply:
Oh, that's fair. I'll update the patch accordingly.